### PR TITLE
Merge Ado's navigation to master

### PR DESCRIPTION
**Pull request by ViliamVadocz**

### CODE REVIEW

**Comment by AdrianJakubcik** [comment]
Btw There is no more need for the include.js in js folder and the navigation.html is not used anymore.

File: js/navigation.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/** This is a debug message that tells me wether or not my javascript was loaded, this debug message can be checked in developer tools in chrome under the tab named console
+ * 
+ * Difficulty: Easy
+*/
```
Comment (DiscreteConcrete) [inline js/navigation.js]: these expanded comments are nicer when you add an extra space before, so that the asterisks are aligned

File: css/nav_style.css [hunk 2]
```diff
@@ -0,0 +1,208 @@
+@import url('https://fonts.googleapis.com/css?family=Didact+Gothic');
+
+/* This is the default style that will be used, in-case there are no media queries this is the default style*/
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: in case is two separate words

Comment (DiscreteConcrete) [inline css/nav_style.css]: let's be a bit cleaner on formatting here: let's put space before the `*/` in comments

File: css/nav_style.css [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+/* I like to start with the mobile viewport first as my base and then just expand on that code using queries etc.*/
+.nav-desk{
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: let's be a bit stricter on formatting here as well: space before `{`

File: css/nav_style.css [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    .nav-desk {
+        display: block;
+    }
+
+    .nav-phone{
+        display: none;
+    }
+
+    #search:hover{
+    background-color:#e9e9e9;
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: indent

File: css/nav_style.css [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+        margin-top: 0.5%;
+        padding: 0;
+        border-bottom: 1px solid black;
+        position: relative;
+        background-color: inherit;
+        width: 99%;
+    }
+
+    #language:hover{
+    background-color: #e7e7e7;
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: indent

File: css/nav_style.css (hunk 3 above)
Comment (DiscreteConcrete) [inline css/nav_style.css]: what does the "desk" mean here? Desktop? the naming is a bit confusing

File: css/nav_style.css [hunk 6]
```diff
@@ -0,0 +1,208 @@
+@import url('https://fonts.googleapis.com/css?family=Didact+Gothic');
+
+/* This is the default style that will be used, in-case there are no media queries this is the default style*/
+/* I like to start with the mobile viewport first as my base and then just expand on that code using queries etc.*/
+.nav-desk{
+    display: block;
+}
+
+.nav-desk > ul{
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: I really like how verbose you are in some parts of the codebase - but here there are some non-obvious properties that we are evoking (like `padding-inline-start`) - can we add some more comments?

File: css/nav_style.css [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    justify-content: space-between;
+}
+
+.nav-phone > .left{
+    position: relative;
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-start;
+    float: left;
+    z-index: 4;
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: why 4?

File: css/nav_style.css [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+body{
+    padding: 0;
+    margin: 0;
+}
+
+.active-nav-btn{
+    background-color: #cccccc !important;
+}
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 740px */
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: it says 670 in the code - so why 740?

File: index.html [hunk 9]
```diff
@@ -1,13 +1,71 @@
...
-		<link rel="shortcut icon" href="favicon.ico" type="image/vnd.microsoft.icon"> 
-		<link rel="icon" href="favicon.ico" type="image/vnd.microsoft.icon">
-	</head>
-	<body>
-		<h1>Galileo School</h1>
-		<p>Work in progress... Check back later!</p>
-	</body>
-</html>
+
+<head>
```
Comment (DiscreteConcrete) [inline index.html:5]: yeh, this is a popular indentation choice - maybe let's add a quick comment that this is deliberate - so those who read this code don't think that we're just lazy with indentation!

File: js/navigation.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+const menuBtnImg = document.getElementById('menu-btn-img');
+var original_attribute_src = menuBtnImg.getAttribute('src');
+var exit_img_src = "images/icons/close.png"; 
+
+/** The same as above only now I'm trying to select a different element (in this case it's the menu itself) 
+ * 
+ * Difficulty: Easy
+*/
+const NavMenuDesk = document.getElementById('nav-desk'); 
+var NavMenu = document.getElementById('Navigation');
```
Comment (DiscreteConcrete) [inline js/navigation.js]: why is one const and the other one var?

File: js/navigation.js [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        if(document.querySelector('.content').classList.contains('blur'))
+            document.querySelector('.content').classList.remove('blur');
+        if (window.getComputedStyle(NavMenu).display == "none") {
+            NavMenu.style.display = "flex";
+        }
+        else
+            return;
+    }
+    else if(window.innerWidth < 670)
+    {
```
Comment (DiscreteConcrete) [inline js/navigation.js]: we're a bit inconsistent here about where the `{` goes - maybe would be nicer on previous line

File: js/navigation.js [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+/** Function that checks the browser size whenever it is resized and evaluates the following statements
+ * 
+ * Difficulty: Harder
+ * @description This function is crucial to get rid of annoying bugs that can occur when transitioning from mobile viewport to desktop viewport. 
+ * Such as when you leave your menu open on a mobile viewport and you extend your browser to desktop viewport the blur effect will still be active, and that's a big NO NO!
+ */
+function check_resize(){
+    if(window.innerWidth >= 670){
+        if(document.querySelector('.content').classList.contains('blur'))
+            document.querySelector('.content').classList.remove('blur');
```
Comment (DiscreteConcrete) [inline js/navigation.js]: we're a bit inconsistent here about usage of `{` and `}` - the general rule of thumb in the industry is that if the `if` or `else` and the effect are on the same single line, we don't use braces. however, when it's on the next line(s), then we always use braces.

File: js/navigation.js [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+/** Function that checks the browser size whenever it is resized and evaluates the following statements
+ * 
+ * Difficulty: Harder
+ * @description This function is crucial to get rid of annoying bugs that can occur when transitioning from mobile viewport to desktop viewport. 
+ * Such as when you leave your menu open on a mobile viewport and you extend your browser to desktop viewport the blur effect will still be active, and that's a big NO NO!
+ */
+function check_resize(){
+    if(window.innerWidth >= 670){
+        if(document.querySelector('.content').classList.contains('blur'))
```
Comment (DiscreteConcrete) [inline js/navigation.js]: we're also inconsistent about space after `if` - in the industry it is more popular to consistently place a space after the `if`

**Review by DiscreteConcrete — CHANGES_REQUESTED** [review]
Very nice!! I added comments mostly regarding formatting and code hygiene.
A general style comment would be that in many companies, there is a preference for a maximum line length of 80 characters - that way the reviews are much nicer and you can view the code in a narrower editor.
We could try to use a "linter" like "ES-Lint" or a tool like "Prettier" for this, but maybe later :) 
I will do a second review - a review of functionality later, but it's looking good.

File: css/nav_style.css (hunk 2 above)
Comment (AdrianJakubcik) [reply css/nav_style.css]: Yeah I wasn't sure 'bout that one so I went with the double case. 🤣

File: js/navigation.js (hunk 10 above)
Comment (AdrianJakubcik) [reply js/navigation.js]: My mistake sorry, fixed it.

File: css/nav_style.css (hunk 8 above)
Comment (AdrianJakubcik) [reply css/nav_style.css]: Originally it was 740 but after a little bit of debuging I thought 670 will be better, and forgot to change the comment. Thank you for pointing out 😄  👍

File: css/nav_style.css (hunk 7 above)
Comment (AdrianJakubcik) [reply css/nav_style.css]: Ohh that was just for a debug purpose, forgot to remove it. Sorry 🤕

File: css/nav_style.css [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+
+.nav-desk {
+    display: block;
+}
+
+.nav-desk>ul {
+    display: none;
+    margin: auto;
+    width: 100vw;
+    /* I will be using vw (view-width) units, this is the measure of the browsers content window width */
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: let's put comments ABOVE the lines on which they comment

File: css/nav_style.css [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+/* I like to start with the mobile viewport first, as it serves as a foundation that I can just expand on using queries etc. */
+
+
+/* nav-desk stands for Navigation-Desktop */
+
+.nav-desk {
+    display: block;
+}
+
+.nav-desk>ul {
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: a more common industry standard is to have spaces around the `>`

File: css/nav_style.css [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+    text-align: center;
+    position: fixed;
+    background-color: #9a0101;
+    z-index: 1;
+    padding: 8vw;
+    padding-left: 0;
+    /* This tells the browser that I don't want any padding at the start of the line in my whole unordered list, without this our items inside the ul would be shifted towards right due to this padding value not being 0 */
+}
+
+div.nav-phone.lang{
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: let's adopt the convention of space before `{`

File: css/nav_style.css [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+    flex-direction: column;
+}
+
+.nav-phone .serach-drop .dropdown {
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+}
+
+.nav-phone.language-dropdown-content {
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: this is inconsistent - maybe let's adopt the convention of spaces between classes (before `.`) in CSS blocks?

File: css/nav_style.css [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+    position: absolute;
+    display: none;
+    top: 10vw;
+    z-index: 1;
+    /*border-radius: 0.25vw;*/
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: do we need to keep this as a comment?

File: css/nav_style.css [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+    font-size: 3.5vw;
+    position: absolute;
+    z-index: 1;
+    opacity: 0;
+    /* Opacity is the measure of how transparent an element and its background should be */
+}
+
+.nav-phone .search-input:focus {
+    opacity: 1 !important;
+    /* !important is a special css attribute that overrides any styles that could possibly clash with it (It's like the Principle of Gymnasium compared to the Teacher of Gymnasium */
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: missing ending `)` in comment
also, in English Gymnasium means a workout/sports gym. High school is a better translation :)

File: css/nav_style.css [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+}
+
+.logo-img-open-nav {
+    margin-bottom: 0.2vw !important;
+    padding-bottom: 0.5vw !important;
+}
+
+.title-open-nav {
+    color: ghostwhite !important;
+    /*margin-left: 2vw !important;*/
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: do we need to keep this?

File: css/nav_style.css [hunk 21]
```diff
@@ -0,0 +1,354 @@
...
+}
+
+.title-cont-open-nav {
+    margin-bottom: 0.2vw !important;
+    padding-bottom: 0.5vw !important;
+}
+
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 670px (px stands for pixels) */
+
```
Comment (DiscreteConcrete) [inline css/nav_style.css:278]: why so many empty lines between them?

File: css/nav_style.css [hunk 22]
```diff
@@ -0,0 +1,354 @@
...
+    /*margin-left: 2vw !important;*/
+}
+
+.title-cont-open-nav {
+    margin-bottom: 0.2vw !important;
+    padding-bottom: 0.5vw !important;
+}
+
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 670px (px stands for pixels) */
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: let's explain how we got to 670 - because the next person working on it might wonder why such exact number

File: css/nav_style.css [hunk 23]
```diff
@@ -0,0 +1,354 @@
...
+    .nav-desk li:hover {
+        background-color: rgba(240, 240, 240, 0.89);
+        color: #4e4e4e;
+        cursor: pointer;
+    }
+    .nav-desk li {
+        margin-bottom: 0 !important;
+        position: relative;
+        color: black;
+        /*border-bottom: 2px solid black;*/
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: do we need to keep this?

File: index.html [hunk 24]
```diff
@@ -1,17 +1,112 @@
...
-		<link rel="shortcut icon" href="favicon.ico" type="image/vnd.microsoft.icon"> 
-		<link rel="icon" href="favicon.ico" type="image/vnd.microsoft.icon">
-		<meta name="copyright" content="© 2019, GALILEO SCHOOL, s.r.o.">
-	</head>
-	<body>
-		<h1>Galileo School</h1>
-		<p>Work in progress... Check back later!</p>
-	</body>
-</html>
+<!--This Indention is Totally Legal and We Are 100% Certain About This-->
```
Comment (DiscreteConcrete) [inline index.html]: why so many capitalised letters? :)

Comment (DiscreteConcrete) [inline index.html]: Also, there tends to be a pedantic convention in the community that there should be a space after `<!--` and before `-->` - and also in CSS after `/*` and before `*/`

File: index.html [hunk 25]
```diff
@@ -1,17 +1,112 @@
...
+
+<head>
+    <!--With this meta tag we are telling the browser what character set we will be using on our web-page (It's crucial for the Slovak diacritic to work, this is because ASCII the default character set doesn't include the weird Slovak symbols like ´ or ô or ˇ etc.) You can read more about this at [https://www.w3schools.com/html/html_charset.asp] -->
+    <meta charset="utf-8">
+    <!-- ADD COMMENT -->
```
Comment (DiscreteConcrete) [inline index.html]: ?

File: index.html [hunk 26]
```diff
@@ -1,17 +1,112 @@
...
+<body>
+    <!--This is the Header where stuff like Logo and Navigation bar can be found (Sometimes you may see some nice front pictures in here for a design reason)-->
+    <!-- Header Start -->
+    <header>
+        <!-- Navigation Bar Starts Here -->
+        <nav>
+            <div id="nav-bar" class="nav-phone">
+                <div class="left">
+                    <div class="nav-phone search-drop dropdown square">
+                        <div id="search" class="nav-phone search"><img id="menu-img" src="images/icons/search.png">
```
Comment (DiscreteConcrete) [inline index.html]: maybe would be nicer if img was on the next line?

File: index.html [hunk 27]
```diff
@@ -1,17 +1,112 @@
...
+                        </div>
+                        <div class="nav-phone search-dropdown-content">
+                            <input class="nav-phone search-input" type="text" placeholder="Search.." name="search">
+                        </div>
+                    </div>
+                    <div id="logo-container" class="nav-phone school-logo-container">
+                        <li class="nav-phone logo"><img id="logo" src="images/logo.jpg"></li>
```
Comment (DiscreteConcrete) [inline index.html]: maybe would be nicer if img had its own line?

File: js/jquery.js [hunk 28]
```diff
@@ -0,0 +1,97 @@
+// This tells the browser to execute certain function as soon as it is ready (as soon as the page is fully loaded)
```
Comment (DiscreteConcrete) [inline js/jquery.js]: `to execute certain function` sounds grammatically awkward - maybe an article would help or maybe it should be plural?

File: js/jquery.js [hunk 29]
```diff
@@ -0,0 +1,97 @@
...
+$(document).ready(function() {
+
+    // Using Id I'm selecting the element with the given Id in this case it is the navigation button
```
Comment (DiscreteConcrete) [inline js/jquery.js]: i feel some punctuation here would help

File: js/jquery.js [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+    var navBtn = '#nav-btn';
+
+    // "Installing" Click event listener on the selected element (When clicked the navigation box slides up and down)
+    $(navBtn).click(function() {
+        $('.navbar').slideToggle('fast');
```
Comment (DiscreteConcrete) [inline js/jquery.js]: there seems to be a lot happening in this function - it's a bit hard to follow!

File: js/jquery.js [hunk 31]
```diff
@@ -0,0 +1,97 @@
...
+
+    // Using Id I'm selecting the element with the given Id in this case it is the language button
+    var langBtn = '#langBtn';
+
+    // "Installing" Click event listener on the selected element (When clicked the language box slides up and down)
+    $(langBtn).click(
+        function() {
+            $('.language-dropdown-content').fadeToggle(200);
+            return;
+        });
```
Comment (DiscreteConcrete) [inline js/jquery.js]: would look nicer if the `);` was on the next line

File: js/jquery.js [hunk 32]
```diff
@@ -0,0 +1,97 @@
...
+            } else {
+                slideOut();
+
+                /** Temporary Solution */
+                searching = false;
+            }
+            return;
+        });
+
+    $(document).click(function() {
```
Comment (DiscreteConcrete) [inline js/jquery.js]: this feels a bit dense

File: js/jquery.js [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+        if (!$('.language-dropdown-content').is(":hover") && !$('.language-dropdown-content').is(":hidden") && !$(langBtn).is(":hover")) {
```
Comment (DiscreteConcrete) [inline js/jquery.js]: we are a little bit inconsistent in using `'` and `"`

File: js/jquery.js [hunk 34]
```diff
@@ -0,0 +1,97 @@
...
+        $(selector).animate({
+            "width": "0",
+            "opacity": 0,
+        }, 400);
+        $(selector).hide(500);
+        $(logo).removeClass('no-display');
+        $(title).removeClass('no-display');
+    };
+
+    //Use the code below for the desktop version (It is the hover version)
```
Comment (DiscreteConcrete) [inline js/jquery.js]: if we need the code below, maybe we can explain why we commented it out for now?

File: js/navigation.js [hunk 35]
```diff
@@ -0,0 +1,110 @@
...
+
+var original_attribute_src = menuBtnImg.getAttribute('src');
+var parsed_dir = Parse_Url(window.location.pathname, 2);
+
+if (parsed_dir == "sk")
+    var exit_img_src = "../images/icons/close.png";
+else
+    var exit_img_src = "images/icons/close.png";
+
+/**
```
Comment (DiscreteConcrete) [inline js/navigation.js]: maybe we're missing an actual explanation of what this does?

File: js/navigation.js [hunk 36]
```diff
@@ -0,0 +1,110 @@
...
+ * 
+ * Difficulty: Easy
+ */
+const menuBtn = document.getElementById("nav-btn");
+const menuBtnImg = document.getElementById('menu-btn-img');
+
+var original_attribute_src = menuBtnImg.getAttribute('src');
+var parsed_dir = Parse_Url(window.location.pathname, 2);
+
+if (parsed_dir == "sk")
```
Comment (DiscreteConcrete) [inline js/navigation.js]: usually, when the if is multiline, the convention is usually to include the `{` and `}`

File: js/navigation.js [hunk 37]
```diff
@@ -0,0 +1,110 @@
...
+ * 
+ * Difficulty: Easy
+ */
+console.log("Javascript Loaded!");
+
+/** Here I'm trying to select a specific element (in this case it is the navigation button) from my html file using its id(identifier) which is unique through-out the document 
+ * 
+ * Difficulty: Easy
+ */
+const menuBtn = document.getElementById("nav-btn");
```
Comment (DiscreteConcrete) [inline js/navigation.js:6]: use of const is interesting: usually people would either use `var` everywhere (old javascript) or `const` and `let` instead (but not `var`). What is our reason to use `const` and `var`? and how much of the world supports the new javascript with `const` and `let` by now?

File: js/navigation.js [hunk 38]
```diff
@@ -0,0 +1,110 @@
+/** This is a debug message that tells me wether or not my javascript was loaded, this debug message can be checked in developer tools in chrome under the tab named console
+ * 
+ * Difficulty: Easy
```
Comment (DiscreteConcrete) [inline js/navigation.js]: do we still need the difficulty indicators?

File: js/navigation.js [hunk 39]
```diff
@@ -0,0 +1,110 @@
...
+/** The same as above but this time I'm actually listening when a browser is being resized. So whenever you extend or shrink your Internet Browser this event will fire up, again completing a function that I've passed into the event.
+ * 
+ * Difficulty: Medium
+ */
+window.addEventListener("resize", check_resize, false);
+
+/** Function that shows and hides our menu on mobile devices
+ * 
+ * Difficulty: Harder
+ * @description What this function does to achieve the effect of hiding and showing our menu after clicking a button on a mobile device,
```
Comment (DiscreteConcrete) [inline js/navigation.js]: I feel this description feels a bit grammatically awkward

Comment (DiscreteConcrete) [inline js/navigation.js]: also, for what reasons do we use the @description here, but not in many other places?

File: js/navigation.js [hunk 40]
```diff
@@ -0,0 +1,110 @@
...
+ */
+function check_resize() {
+    if (window.innerWidth >= 670) {
+        if (document.querySelector('.content').classList.contains('blur')) {
+            document.querySelector('.content').classList.remove('blur');
+        }
+        if (menuBtnImg.getAttribute('src') == exit_img_src) {
+            menuBtnImg.setAttribute('src', original_attribute_src);
+        }
+        if (window.getComputedStyle(NavMenu).display == "none" || window.getComputedStyle(NavMenu).display == "block") {
```
Comment (DiscreteConcrete) [inline js/navigation.js:87]: I like this entire function - but maybe we could add a tiny bit more explanation?

File: js/navigation.js [hunk 41]
```diff
@@ -0,0 +1,110 @@
...
+            return;
+        }
+    } else if (window.innerWidth < 670) {
+        if (window.getComputedStyle(NavMenu).display == "flex") {
+            NavMenu.style.display = "none";
+        }
+    }
+}
+
+function resetMenu() {
```
Comment (DiscreteConcrete) [inline js/navigation.js:106]: maybe we can explain what this does?

**Review by DiscreteConcrete — CHANGES_REQUESTED** [review]

File: js/navigation.js (hunk 37 above)
Comment (AdrianJakubcik) [reply js/navigation.js:6]: I decided to use `const` for things that are constantly the same no matter what. Like html elements, when I select any html element I know it won't be changing anytime soon, thus I assign it as a `const`. But things that may change throughout the code are assigned as `var` meaning variable. Variable is something that may change because it is vary-able or able to vary. It's totally just my way of dividing constants and non-constants. Btw the support for `const` is around `98.97%`.

File: js/navigation.js (hunk 38 above)
Comment (AdrianJakubcik) [reply js/navigation.js]: I'm leaving them there for the others, if they would like to start with javascript I'm trying to show them what things are easier to do and which are harder to do.

File: js/navigation.js (hunk 39 above)
Comment (AdrianJakubcik) [reply js/navigation.js]: As I was playing a little bit with Js it seems that Javascript only allows to add description to function, class or any other object that can be called during script execution more than once. Meaning I can't add description to code that is not enclosed and turned into a 'static' object, it's not static but I can't find the right word for it atm. This is the reason why not many things have `@description` but most of my function documentations do include a `@description`.

File: js/jquery.js (hunk 31 above)
Comment (AdrianJakubcik) [reply js/jquery.js]: My formatter always moves it back after the `}`. I'm sorry for that.

File: js/navigation.js (hunk 38 above)
Comment (ViliamVadocz) [reply js/navigation.js]: I think they just make the code messier.

File: css/nav_style.css (hunk 17 above)
Comment (ViliamVadocz) [reply css/nav_style.css]: I think that has a different meaning based on this thread: https://stackoverflow.com/questions/5116288/how-to-select-an-element-with-2-classes

File: js/navigation.js (hunk 38 above)
Comment (AdrianJakubcik) [reply js/navigation.js]: As long as I know the point of a documentation is to include any helpful information that would explain the purpose of a function and/or the way it works. By helpful information I mean anything that can help others understand the function. By telling others how difficult a certain code may be, can give them a way to compare their abilities. Making it easier to choose the most suitable task for them. In our case I do include some unnecessary information but bear in mind I'm not doing it for myself, but for the others.

File: css/nav_style.css (hunk 17 above)
Comment (DiscreteConcrete) [reply css/nav_style.css]: completely missed that!

File: js/navigation.js (hunk 37 above)
Comment (DiscreteConcrete) [reply js/navigation.js:6]: Good! Now what reasons do we have to use `var` over `let`?

File: js/navigation.js (hunk 38 above)
Comment (DiscreteConcrete) [reply js/navigation.js]: I agree with the general point of helping others, but I am sure there is a level where we would be over-commenting (for example if I was to talk about what I had for lunch while coding those parts of code). In this particular case, I feel like we are slightly overstepping the line and it is a bit unnecessary - in other words I don't think there is that much value and giving so many lines of code to something with so little value feels like overcommenting

File: js/navigation.js (hunk 39 above)
Comment (DiscreteConcrete) [reply js/navigation.js]: fair!

File: css/nav_style.css [hunk 42]
```diff
@@ -0,0 +1,362 @@
...
+    font-size: 3.5vw;
+    position: absolute;
+    z-index: 1;
+    /* Opacity is the measure of how transparent an element and its background are */
+    opacity: 0;
+}
+
+.nav-phone .search-input:focus {
+    opacity: 1 !important;
+    /* !important is a special css attribute that overrides any styles that could possibly clash with it (It's like the Principle of High School compared to the Teacher of High School) */
```
Comment (DiscreteConcrete) [inline css/nav_style.css]: move comment above what interests us

File: css/nav_style.css (hunk 21 above)
Comment (DiscreteConcrete) [reply css/nav_style.css:278]: this is still unresolved...

File: js/jquery.js (hunk 31 above)
Comment (DiscreteConcrete) [reply js/jquery.js]: not convinced

File: js/navigation.js (hunk 41 above)
Comment (DiscreteConcrete) [reply js/navigation.js:106]: I think I am still a bit confused about this function - I understand what it does, but if I change something in the navbar, how will I know I have to edit something here as well? This has a high likelihood of breaking in the future! What can we do about that?

**Review by DiscreteConcrete — CHANGES_REQUESTED** [review]

File: css/nav_style.css (hunk 19 above)
Comment (ViliamVadocz) [reply css/nav_style.css]: also note principle  != principal

**Comment by ViliamVadocz** [comment]
Would be nice if we resolved already solved comments so it is easier to see what needs addressing.

File: js/navigation.js (hunk 41 above)
Comment (ViliamVadocz) [reply js/navigation.js:106]: Maybe we could reuse the function found in `jquery.js` which activates on click to toggle the menu visibility which would reduce the amount of code that needs to be updated if anything changes.

**Review by DiscreteConcrete — APPROVED** [review]
